### ISSUE TITLE
Workspace nav labels and page titles show the directory name, not the front-matter title

### ISSUE THREAD

**Issue by olegshulyakov** [TokenCemetery/teach#49]
## What is wrong

Eight of the nineteen workspaces are labelled in the sidebar by their directory name instead of by their front-matter `title`. The same wrong string reaches the overview page's `<title>`, its `og:title` and `twitter:title`, and the visible page heading the theme renders from `page.title`.

| Directory | Sidebar and page title | Front-matter `title` |
|---|---|---|
| `llm/finetuning` | Finetuning | Adapter Fine-Tuning |
| `llm/rag` | Rag | RAG |
| `architecture/api-design` | Api design | API Design |
| `architecture/distributed-systems` | Distributed systems | Distributed Systems |
| `programming/csharp` | Csharp | C# |
| `programming/golang` | Golang | Go |
| `programming/sql` | Sql | SQL |
| `programming/typescript` | Typescript | TypeScript |

The other eleven look right only because the capitalized directory name happens to equal the title.

## Steps to reproduce

```bash
.venv/bin/mkdocs build --strict
grep -o "<title>[^<]*</title>" site/programming/csharp/index.html
```

Prints `<title>Csharp - teach</title>`, where `learning/programming/csharp/README.md` declares `title: C#`.

## Cause

`mkdocs-section-index` folds the `README.md` overview into its section label, which is what `mkdocs.yml` enables it for, and while doing so it assigns the section's own title to the page:

```python
page.title = section.title
```

The section title is derived from the directory name, and `Page.title` is a `weak_property`, so that instance attribute permanently shadows the front matter. The plugin does this because for a hand-written `nav` the section title is the authored one; here navigation is generated, so the directory name is all the section has.

This contradicts `.agents/skills/teach/PUBLISHING.md`, which states that `title` is what the sidebar and the browser tab render.

## Done when

- Every workspace's sidebar label, `<title>`, social-card titles and page heading come from its front-matter `title`.
- Section labels that have no page behind them (`Programming`, `Lessons`, `Reference`) keep their directory-derived names.
- `.venv/bin/mkdocs build --strict` stays clean.

## Files involved

- `hooks.py`: the `on_nav` hook added for #48 already runs after the plugin and already restores a shadowed `title` for the sections it unfolds. The folded overview pages need the same one line.
